crates/revm/src/evm_impl.rs

```diff
@@ -241,73 +335,11 @@ impl<'a, GSPEC: Spec, DB: Database, const INSPECT: bool> Transact<DB::Error>
-                    //   - Deposit transactions (non-system) report their gas limit as the usage.
-                    //     No refunds.
-                    //   - Deposit transactions (system) report 0 gas used. No refunds.
-                    //   - Regular transactions report gas usage as normal.
-                    // - Regolith (success path):
-                    //   - Deposit transactions (all) report their gas used as normal. Refunds
-                    //     enabled.
-                    //   - Regular transactions report their gas used as normal.
-                    #[cfg(feature = "optimism")]
-                    if self.data.env.cfg.optimism
-                        && (!is_deposit || GSPEC::enabled(SpecId::REGOLITH))
-                    {
-                        // For regular transactions prior to Regolith and all transactions after
-                        // Regolith, gas is reported as normal.
-                        gas.erase_cost(ret_gas.remaining());
-                        gas.record_refund(ret_gas.refunded());
-                    } else if is_deposit && tx_system.unwrap_or(false) {
-                        // System transactions were a special type of deposit transaction in
-                        // the Bedrock hardfork that did not incur any gas costs.
-                        gas.erase_cost(tx_gas_limit);
-                    }
-                }
-                return_revert!() => {
-                    #[cfg(not(feature = "optimism"))]
-                    gas.erase_cost(ret_gas.remaining());
-
-                    // On Optimism, deposit transactions report gas usage uniquely to other
-                    // transactions due to them being pre-paid on L1.
-                    //
-                    // Hardfork Behavior:
-                    // - Bedrock (revert path):
-                    //   - Deposit transactions (all) report the gas limit as the amount of gas
-                    //     used on failure. No refunds.
-                    //   - Regular transactions receive a refund on remaining gas as normal.
-                    // - Regolith (revert path):
-                    //   - Deposit transactions (all) report the actual gas used as the amount of
-                    //     gas used on failure. Refunds on remaining gas enabled.
-                    //   - Regular transactions receive a refund on remaining gas as normal.
-                    #[cfg(feature = "optimism")]
-                    if self.data.env.cfg.optimism
-                        && (!is_deposit || GSPEC::enabled(SpecId::REGOLITH))
-                    {
-                        gas.erase_cost(ret_gas.remaining());
-                    }
-                }
-                _ => {}
-            }
-        }
+        // Spend the gas limit. Gas is reimbursed when the tx returns successfully.
+        #[cfg(not(feature = "optimism"))]
+        let gas = self.use_gas(tx_gas_limit, ret_gas, exit_reason)?;
```

Comment:
nit: The if `crate::USE_GAS` condition can be pulled out of the `use_gas` impls and placed around these invocations.

Reply:
That's cool by me